lib/textris/phone_formatter.rb

```diff
@@ -0,0 +1,27 @@
+module Textris
+  class PhoneFormatter
+    class << self
+      def format(phone = '')
+        return phone if is_a_short_code?(phone) || is_alphameric?(phone)
+        "#{'+' unless phone.start_with?('+')}#{phone}"
+      end
+
+      # Short codes have more dependencies and limitations;
+      # but this is a good general start
+      def is_a_short_code?(phone)
+        !!phone.to_s.match(/\A\d{4,6}\z/)
+      end
+
+      def is_a_phone_number?(phone)
+        Phony.plausible?(phone)
+      end
+
+      # We ASSUME that if someone passes a value
+      # that cannot be resolved as a short code or a phone
+      # number, then it is Alphameric Sender ID
+      def is_alphameric?(phone)
+        !is_a_phone_number?(phone) && !is_a_short_code?(phone)
```

Comment:
[The rules for an alphanumeric ID](https://support.twilio.com/hc/en-us/articles/223181348-Getting-started-with-Alphanumeric-Sender-ID) are:

> You may use any combination of 1 to 11 letters, A-Z and numbers, 0-9. Both lowercase and uppercase characters are supported as well as spaces. 1 letter and no more than 11 alphanumeric characters may be used.

I messed around with a regex to match this constraint and came up with:

```
\A(?=.*[a-zA-Z])[a-zA-z\d]{1,11}\z
```

Broken down, this is:

```
\A               # Start of the string
(?=.*[a-zA-Z])   # Lookahead to ensure there is at least one letter in the entire string
[a-zA-z\d]{1,11} # Between 1 and 11 characters in the string
\z               # End of the string
```

So we could rewrite this as:

```ruby
def is_alphanumeric?(phone)
  !!phone.to_s.match(/\A(?=.*[a-zA-Z])[a-zA-z\d]{1,11}\z/)
end
```

Reply:
Awesome!